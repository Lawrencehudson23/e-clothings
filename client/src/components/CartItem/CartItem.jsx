import React from "react";

import "./CartItem.scss";
const CartItem = ({ item }) => {
  const { imageUrl, name, price, quantity } = item;
  return (
    <div className="cart-item">
      <img src={imageUrl} alt="item" />
      <div className="item-details">
        <span className="name">{name}</span>
        <span className="price">
          {quantity} X <span className="dollar">$</span>
          {price}
        </span>
      </div>
    </div>
  );
};

export default CartItem;
