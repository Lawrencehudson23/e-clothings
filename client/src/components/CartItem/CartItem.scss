.cart-item {
    width: 100%;
    display: flex;
    height: 80px;
    margin-bottom: 15px;

    img {
        width: 30%;
    }

    .item-details {
        width: 70%;
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        justify-content: center;
        padding: 10px 20px;
        font-size: 15px;

        .dollar {
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
            font-weight: 700;
        }
    }
}