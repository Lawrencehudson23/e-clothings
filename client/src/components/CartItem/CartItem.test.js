import React from "react";
import { shallow } from "enzyme";
import CartItem from "./CartItem";

it("should render CartItem component", () => {
  const mockItem = {
    imageUrl: "www.testImage.com",
    name: "hats",
    price: 10,
    quantity: 2,
  };
  expect(shallow(<CartItem item={mockItem} />).debug()).toMatchSnapshot();
});
