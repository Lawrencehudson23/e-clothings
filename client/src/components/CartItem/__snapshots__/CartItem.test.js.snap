// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartItem component 1`] = `
"<div className=\\"cart-item\\">
  <img src=\\"www.testImage.com\\" alt=\\"item\\" />
  <div className=\\"item-details\\">
    <span className=\\"name\\">
      hats
    </span>
    <span className=\\"price\\">
      2
       X 
      <span className=\\"dollar\\">
        $
      </span>
      10
    </span>
  </div>
</div>"
`;
